### PR TITLE
Update e2e tests to reflect that drafts are now part of post status

### LINKED ISSUE #90290: Gutenberg: Update e2e tests to reflect that drafts are now part of post status
The UI for unpublishing a post has https://github.com/WordPress/gutenberg/pull/60456 such that you now have to interact with the post status panel pop out. Also, the Update CTA text has changed to save

This is currently breaking some nightly e2e tests for Gutenberg. This change modifies the e2e's expectations to fit the new realities.

### PR DESCRIPTION
**Pull request by olaseni**
Fixes: https://github.com/Automattic/wp-calypso/issues/90290
Related: https://github.com/WordPress/gutenberg/pull/60456

The UI for unpublishing a post has [changed](https://github.com/WordPress/gutenberg/pull/60456) such that you now have to interact with the post status panel pop out. Also, the `Update` CTA text has changed to `save`

This is currently breaking some nightly e2e tests for Gutenberg. This change modifies the e2e's expectations to fit the new realities.

Testing instructions
---

[Running e2e tests for Calypso](https://github.com/Automattic/wp-calypso/tree/trunk/test/e2e)

Run the page basic flow e2e edge tests for Simple and AT sites, which were failing before this fix:

```bash
cd test/e2e
yarn jest --clearCache && yarn build 
GUTENBERG_EDGE=1 yarn jest specs/editor/editor__post-advanced-flow.ts
GUTENBERG_EDGE=1 TEST_ON_ATOMIC=true yarn jest specs/editor/editor__post-advanced-flow.ts
```

### CODE REVIEW

File: packages/calypso-e2e/src/lib/pages/editor-page.ts [hunk 1]
```diff
@@ -769,10 +769,13 @@ export class EditorPage {
...
 	async unpublish(): Promise< void > {
 		const editorParent = await this.editor.parent();
-
 		await this.editorToolbarComponent.switchToDraft();
 		// @TODO: eventually refactor this out to a ConfirmationDialogComponent.
-		await editorParent.getByRole( 'button' ).getByText( 'OK' ).click();
+		// Saves the draft
+		await Promise.race( [
+			this.editorToolbarComponent.clickPublish(),
+			editorParent.getByRole( 'button' ).getByText( 'OK' ).click(),
```
Comment (fredrikekelund) [inline packages/calypso-e2e/src/lib/pages/editor-page.ts:777]: Why do we need both these calls? They seem to accomplish the same thing 🤔

The test also seems to complete successfully (on Simple and on Atomic) with only one of these lines.

**Review by fredrikekelund — COMMENTED** [review]
Changes make sense and work as expected 👍 

I left one comment that would be good to resolve before landing, though

File: packages/calypso-e2e/src/lib/pages/editor-page.ts (hunk 1 above)
Comment (olaseni) [reply packages/calypso-e2e/src/lib/pages/editor-page.ts:777]: Actually the edge versions on simple and AT are the same and you won't notice a difference there. There are both running 18.2.0.

For versions before i.e. 18.1.2, which is what ordinary sites ( non-edge sites ) are using.

You can test this with:

`yarn jest specs/editor/editor__post-advanced-flow.ts`

... without the `GUTENBERG_EDGE=1` or  `TEST_ON_ATOMIC=true`. Apologies for not leaving this in the description.

The UI for that version is different. That OK button is only present there.

Screenshot 2024-05-06 at 10 48 09

Without using `Promise.race()` with those two lines, you'll get this error:

```

  ● Editor: Advanced Post Flow › Revert post to draft › Switch to draft

    locator.click: Timeout 10000ms exceeded.
    =========================== logs ===========================
    waiting for frameLocator('iframe[src*="calypsoify"]').locator('body.block-editor-page').locator('.interface-navigable-region[class*="header"] button.editor-post-publish-button__button[aria-disabled="false"]')
    ============================================================

      276 | 		const editorParent = await this.editor.parent();
      277 | 		const publishButtonLocator = editorParent.locator( selectors.publishButton( 'enabled' ) );
    > 278 | 		await publishButtonLocator.click();
          | 		                           ^
      279 | 	}
      280 |
      281 | 	/**

      at EditorToolbarComponent.click [as clickPublish] (../../packages/calypso-e2e/src/lib/components/editor-toolbar-component.ts:278:30)
      at EditorPage.unpublish (../../packages/calypso-e2e/src/lib/pages/editor-page.ts:775:3)
      at Object.<anonymous> (specs/editor/editor__post-advanced-flow.ts:155:4)


```

Comment (fredrikekelund) [reply packages/calypso-e2e/src/lib/pages/editor-page.ts:777]: Gotcha! I think I simply forgot to build after making my changes to the tests, or they wouldn't have passed 🤦‍♂️

**Review by fredrikekelund — APPROVED** [review]
LGTM 👍

**Pull request merged**